src/colinajose/service/GradesService.java

```diff
@@ -0,0 +1,114 @@
+package colinajose.service;
+
+import colinajose.model.people.Student;
+import colinajose.model.schoolEntities.Course;
+import colinajose.model.schoolEntities.Grade;
+import colinajose.model.schoolEntities.GradingScale;
+import colinajose.model.schoolEntities.Kardex;
+import colinajose.model.schoolEntities.Subject;
+import datastructures.arraylist.MyArrayList;
+import datastructures.circulardoublylinkedlist.MyCircularDoublyLinkedList;
+
+public class GradesService {
+
+    public void setGradingScale(Course course, double scholarship, double expelled, double notify) {
+        GradingScale gradingScale = new GradingScale(expelled, notify, scholarship);
+        course.setGradingScale(gradingScale);
+    }
+
+    public void addGrade(double grade, Student student, Subject subject) {
+        Grade studentGrade = new Grade(student, grade);
+        subject.addGrade(studentGrade);
+    }
+
+    public MyCircularDoublyLinkedList<Grade> computeGrades(Kardex kardex) {
+        MyCircularDoublyLinkedList<Student> students = kardex.getStudents();
+        MyCircularDoublyLinkedList<Subject> subjects = kardex.getCourse().getSubjects();
+        MyCircularDoublyLinkedList<Grade> averageGrades = new MyCircularDoublyLinkedList<>();
+        for (int i = 0; i < students.size(); i++) {
+            Student student = students.get(i);
+            averageGrades.add(computeGrade(subjects, student));
+        }
+        return averageGrades;
+    }
+
+    private Grade computeGrade(MyCircularDoublyLinkedList<Subject> subjects, Student student) {
+        double average = 0;
+        for (int i = 0; i < subjects.size(); i++) {
+            MyCircularDoublyLinkedList<Grade> grades = subjects.get(i).getGrades();
+            for (int j = 0; j < grades.size(); j++) {
+                Grade grade = grades.get(j);
+                if (grade.getStudent().equals(student)) {
+                    average += grade.getFinalGrade();
+                    break;
+                }
+            }
+        }
+        average /= subjects.size();
+        return new Grade(student, average);
+    }
+
+    public void computeStudentStates(Kardex kardex, MyCircularDoublyLinkedList<Grade> averageGrades) {
```

Comment:
This is were I want you to think how can you use the DataStructures you learned.
Are Lists the best way to find out which state the student should have?
Let's assume that the default state value for the students is Average. So you don't really care about them.
But with design you are looping again to all of the students of the class even if they don't are meaningful for this stage.
Wouldn't be great that you can compute this while calculating the averages?